### PR TITLE
Tweaks voice analyzers, pulsing a grenade's bomb wire now uses the grenade's timer delay.

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Makes suicide bombing with voice analyzers harder, while keeping the more fun behavior by adding some tells and delays.
Things attached to a grenade's pulse wire use the grenades timer, rather then just insta booming.

## Why It's Good For The Game

The analyzer brings some really fun things to the game, IE: don't say ____ memes, repeating wgw over the radio as the station blows to smithereens, ect.

The trouble is that's not how it's used in the main. It's main use currently is to suicide bomb when you get accosted by an antag.

I don't think we should have features that push towards/allow "you killed me, die" with this degree of control. There's a niche there in the health sensor, but that's uncontrolled, and therefore more of a risk.

So lets add some tells for the thing, and slow it down even more. As it currently stands it'll boom a grenade after at minimum 5 seconds, unless some fuck added a way to lower that timer to 0, and will trigger everything else after 2 seconds. Just long enough for the WHIRRRRRRRRRRRR CLUNK to play. I love the whirrrrrrrrrrrrrrrrrr.

I've also added a new parent folder for odd sound licenses. I love me a good whirrrrrr.
Whir sourced from a portion of https://freesound.org/people/engreitz/sounds/79021/
## Changelog
:cl:
tweak: If you attach something to a grenade's boom wire it won't instantly trigger on pulse, instead using the same timer used for inhand activations.
add: The voice analyzer now says something when it triggers, and plays a short whirrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrr sound. The time between parse and trigger has been increased to 2 seconds as well. Oh and they can't be used when whispering.
/:cl:

### CODE REVIEW

**Comment by trexter555** [comment]
i dead pr

**Comment by Couls** [comment]
I support you in this battle

**Comment by LemonInTheDark** [comment]
> 
> 
> i dead pr

This is the last time I played as an antag 
image
This is the last time I died to a voice activated bomb iirc 
image

**Comment by necromanceranne** [comment]
If people like to think this wasn't an issue, the in vogue thing for a time was to random name random face and voice bomb people who were known for being good as antags when they rolled it. Literally sitting on the wings waiting to pop it on them. We really do have some pretty dreadful people about.

It is however a massive shame to remove these, since it's a fun item in other ways.

**Comment by Investigator77** [comment]
Not a big fan tbh, sounds more like a player problem than a mechanical problem.

**Comment by Ratvar** [comment]
I am not sure if there is any reason to keep health analyzer if voice analyzer goes. Yes, it is a bit more uncontrollable, but it does the exact same thing - suicidebomb, and so will replace voice analyzer in this situation, with no change on the outcome.

**Comment by LemonInTheDark** [comment]
If the admins can't fix it player problems become mechanical problems

**Comment by LemonInTheDark** [comment]
> 
> 
> I am not sure if there is any reason to keep health analyzer if voice analyzer goes. Yes, it is a bit more uncontrollable, but it does the exact same thing - suicidebomb, and so will replace voice analyzer in this situation, with no change on the outcome.

It's a wild ride we can't get off

**Comment by Ratvar** [comment]
We actually easily can.

**Comment by CRITAWAKETS** [comment]
Want to suicide bomb? Use a health analyzer. If you are the type to suicide bomb you dont care about collateral anyway.

Want to activate the bomb at any time? Use a remote signaler.

**Comment by zxaber** [comment]
The issue could be *somewhat* alleviated by making whispering unable to set the voice analyzer off. Please speak clearly and loudly into the device.

At least then you'd have to use it sometime before crit.

**Comment by necromanceranne** [comment]
Actually @CRITAWAKETS I found that the presumed precision of those bombs granted by voice activation is what justified a lot of it. I've seen a few people argue that since it wasn't a health bomb it's just standard validhunting, even if it is punching a massive hole in the station every time, because only them and their target were gibbed. That's why they did it. It was a deliberate attempt to ride the line to use a no answer move.

If this is still an issue despite me pointing this out in previous terms to admins while an admin, then yes, gut it. 

The issue is really that bombs are kind of fucking powerful. It has always been thus far that the only balancing element to explosions are their delivery systems. Ttvs are big and bulky but you can game it. Bomb core+c4 is notorious because it is portable wide bombs with grenade timers at no effort but most of your tc cost (though often used to suicide bomb anyway, more lame and uninspired than an issue since antags are involved). And we all know about grenades. And cryo syringes were removed explicitly because they were just rocket launchers. A lot of the precision explosions are antagonist tools, basically. 

Having some degree of control like this seems to trick players into thinking that it's fine to do it like any other big highly volitile weapon in an act of self sacrifice, while not actually beholden to those same rules. We do have one other weapon that does this, the explosive lance, but that involves actually hitting the person in question and cognitively you can make that distinction that this is a dangerous weapon with collateral. I see elances far less today now that they aren't throwable and so less reliable. It's that unreliability that makes people hesitate.

**Comment by Jared-Fogle** [comment]
I can get behind the reasoning behind this. I just hope this doesn't mean people who do this will switch to health sensor nades and deal significantly more damage.

**Comment by Istoprocent** [comment]
Don't remove voice analyzers. Simply change signalers to start default activation sequence and not work as an instant activation method.

For example - grenade going off 3 or 5 seconds after voice command (depending on how the timer has been adjusted) not blowing up instantly and so on.

**Comment by Tlaltecuhtli** [comment]
do what isto says, removing voice just means people will use health analyzers and also freely grief with it as now they have the excuse of "couldnt turn off my suicide bomb in time"

**Comment by Jared-Fogle** [comment]
I'm not sure people would even use voice analyzers anymore if that was the case, I guess if you couldn't hear the sound of the grenade.

**Comment by Angustmeta** [comment]
Don't remove a fun primary mechanic like this. Voice analyzer is one of the goddamn main selling points of the game. Making voice activated airlock bolters is one of my favourite things to do. 

Just go with the obvious solution that Isto and others have suggested.

**Comment by Time-Green** [comment]
Adding a delay isn't gonna do much. It just means everyone is gibbed 3-5 seconds later. 

The biggest issue is people using it with bombs to instantly kill antagonists with no counter-play. A good example is people setting the voice activated message to the cult conversion message and then trying to get converted to instantly kill all of cult. 

I think an obvious audible warning that a voice analyzer has been activated before triggering 3-5 seconds later would give traitors a chance to run and cultists a chance to teleport them away.

**Comment by Jared-Fogle** [comment]
> I think an obvious audible warning that a voice analyzer has been activated before triggering 3-5 seconds later would give traitors a chance to run and cultists a chance to teleport them away.

If you're saying this as a positive, I think this just means that nobody will use them anymore. Unless this is only for grenades, in which case people will stop using them for that but fun things like password protected airlocks will still work.

**Comment by Time-Green** [comment]
> > I think an obvious audible warning that a voice analyzer has been activated before triggering 3-5 seconds later would give traitors a chance to run and cultists a chance to teleport them away.
> 
> If you're saying this as a positive, I think this just means that nobody will use them anymore. Unless this is only for grenades, in which case people will stop using them for that but fun things like password protected airlocks will still work.

I meant voice analyzers, but giving all relevant bombs an audible warning and a few seconds to respond to it could make it a lot easier for people to deal with braindead suicide bombers. Grenades already do this (assuming you didnt change the trigger)

**Comment by ATH1909** [comment]
Why not add the delay to grenades instead of to voice analyzers? Like, make it so that when a grenade's boom wire is pulsed, it goes off 3 seconds (or 5 seconds, depending on how you set it up) later (and it'll of course beep loudly before it goes off). That way, you also nip health analyzer grenades in the bud AND also DON'T force voice-activated doors and such to have annoying 3 second delays.

**Comment by FloranOtten** [comment]
If this is such a massive problem (though I haven't often seen it, and the fact that you died last to one 5 months ago attests to that), just make it not fit grenades. Then we get to keep password airlocks, for example.

**Comment by CRITAWAKETS** [comment]
grenades should beep to activation, bombs should tick to activation

**Comment by necromanceranne** [comment]
Who actually uses password airlocks? It's almost always used in bombcraft. All the signaller assemblies are.

**Comment by Jared-Fogle** [comment]
Creative traitors. I think Korol did it last week.

**Comment by Angustmeta** [comment]
> Who actually uses password airlocks? 

I do

**Comment by Angustmeta** [comment]
I use voice analyzers for non-lethal grenades as well. The option should stay, regardless how it's "almost always" used. 

Also nobody in this thread has suggested a silent delay. The suggestion is to make it prime (loudly).

**Comment by IndieanaJones** [comment]
Correct me if I'm wrong, but nanites also are capable of this exact thing, aren't they?

**Comment by Winterous** [comment]
> 
> 
> Correct me if I'm wrong, but nanites also are capable of this exact thing, aren't they?

Nanites can't signal things outside of nanites, like circuits could.

I think that making signals prime the grenade instead of instantly detonating is a really good compromise.

If they do get removed; perhaps we could add something new to serve a similar purpose. I've always liked the tooth implant pills, maybe we could have a tooth implant signaler, allowing you to activate them as long as you're conscious, but requiring the assistance of another person to put it in your tooth.

**Comment by itseasytosee** [comment]
Personally, my only experience is making a lube smoke grenade as a traitor and escaping sec handcuffed. I don't think this niche is filled otherwise.
Tooth implant signaler, or "SUPER REMOTE" signaler brain implants were something I've had on my mind for a while.

**Comment by optimumtact** [comment]
dude wat lmao

**Comment by LemonInTheDark** [comment]
> 
> 
> Why not add the delay to grenades instead of to voice analyzers? Like, make it so that when a grenade's boom wire is pulsed, it goes off 3 seconds (or 5 seconds, depending on how you set it up) later (and it'll of course beep loudly before it goes off). That way, you nip health analyzer grenades in the bud as well AND don't force voice-activated doors and such to have annoying 3 second delays.

That sounds like a good idea, not totally sure about it because it retains the main issue I have with these things, the use as a better suicide bomb, but I'll take a look.

**Comment by optimumtact** [comment]
make the voice analyzer have a visible activation message and a delay upon activating, while it processes the voice.

**Comment by LemonInTheDark** [comment]
> 
> 
> If this is such a massive problem (though I haven't often seen it, and the fact that you died last to one 5 months ago attests to that), just make it not fit grenades. Then we get to keep password airlocks, for example.

My point is to show how little I play, I'll expand on it now. I main sec these days, so I am quite literally immune to the use of these things I'm bothered by, the "you killed me, die" aspect.

That might work..... I'll keep it in mind.

**Comment by LemonInTheDark** [comment]
> 
> 
> The issue could be _somewhat_ alleviated by making whispering unable to set the voice analyzer off. Please speak clearly and loudly into the device.
> 
> At least then you'd have to use it sometime before crit.

This is a good idea in general, I'll toss it in.

**Comment by LemonInTheDark** [comment]
> 
> 
> make the voice analyzer have a visible activation message and a delay upon activating, while it processes the voice.

Sounds about right.

**Comment by itseasytosee** [comment]
I don't see the problem. Are you saying they are redundant because of health analyzers or that they are too powerful because they have advantages over health analyzers?

**Comment by LemonInTheDark** [comment]
I dislike the mindset they feed into. This feeding is caused by the greater control you have with them.

**Comment by itseasytosee** [comment]
what mindset is that, the idea that you could make a suicide bomb?

**Comment by optimumtact** [comment]
It's not that you can make a suicide bomb, it's that you can choose when it goes off, as opposed to the health sensor.

**Comment by Angustmeta** [comment]
> 

> dude wat lmao

It was a memorable part of the Sseth vid

**Comment by Misha600** [comment]
Welcome to /tg/station, where we remove, ruin, change and nerf things for absolutely no reason.

**Comment by LemonInTheDark** [comment]
I expect a 100% ROI on the most precious downthumbs applied to this pr

**Comment by Winterous** [comment]
Why does it say "Reconfiguring transistors"?
Also whispering no longer triggers, big rip.

**Comment by Winterous** [comment]
So now someone needs to add tooth-implant signallers.

I'm just imagining drilling a tiny little hole in a tooth, and cramming the entire walkie talkie thing in there.

**Comment by ExcessiveUseOfCobblestone** [comment]
mfw voice activated signalers become the new singularity

**Comment by ArcaneMusic** [comment]
With the changes, I can thoroughly get behind this.

**Comment by LemonInTheDark** [comment]
> 
> 
> Why does it say "Reconfiguring transistors"?

Cause whirrrrrrrrrrrrrrrrrrr and also I'm bad at scifi names.
> Also whispering no longer triggers, big rip.

Feature

**Comment by LemonInTheDark** [comment]
> 
> 
> So now someone needs to add tooth-implant signallers. (I'm gonna have a crack at it)
> 
> I'm just imagining drilling a tiny little hole in a tooth, and cramming the entire walkie talkie thing in there.

Give it a shot, why not.

**Comment by OmegaShoots** [comment]
Very sad that this comes dangerously close in time to the round that someone suicide bombed the brig with a voice nade.

**Comment by LemonInTheDark** [comment]
bruh my ckey is my username go look it up

**Comment by Winterous** [comment]
> Cause whirrrrrrrrrrrrrrrrrrr and also I'm bad at scifi names.
I recommend a simple "pings gently' or something like that. It's a voice ANALYSER, not a voice TALKER.

**Comment by LemonInTheDark** [comment]
Mmmm actually...

**Comment by Tlaltecuhtli** [comment]
remove the preprime change it breaks proxysensors and mouse traps making them useless

**Comment by Jared-Fogle** [comment]
+1 for making sure mouse trap bombs work, they're extremely fun.

**Comment by Winterous** [comment]
> 
> 
> remove the preprime change it breaks proxysensors and mouse traps making them useless



> 
> 
> +1 for making sure mouse trap bombs work, they're extremely fun.

Getting mixed messages here.

Also pretty sure that proxy GRENADES use snowflake code so this shouldn't apply to them.

**Comment by Istoprocent** [comment]
Change all signalers to have a 3 second delay. Even mousetraps and proxy signalers work fine after this change, since once the process starts it runs its course and the bomb will go off.

Optional have a custom tune play once a signaler is activated such as that one popular nasheed people were raving about a few years ago.

Directly pinging the wire with a multitool would make a bomb go off instantly.

Introduce syndicate signalers that can be adjusted between instant and 3 second delay.

File: code/modules/assembly/voice.dm [hunk 1]
```diff
@@ -22,13 +22,16 @@
 								 "voice sensor")
 	drop_sound = 'sound/items/handling/component_drop.ogg'
 	pickup_sound =  'sound/items/handling/component_pickup.ogg'
+	var/volume = 40
```
Comment (Rohesie) [inline code/modules/assembly/voice.dm:25]: doc

File: code/modules/assembly/voice.dm [hunk 2]
```diff
@@ -76,6 +79,11 @@
 			if(length(raw_message))
 				. = TRUE
 
+/obj/item/assembly/voice/proc/send_pulse()
+	visible_message("clicks", visible_message_flags = EMOTE_MESSAGE)
+	playsound(src, 'licensed_sound/Sampling-Plus-1.0/whirthunk.ogg', 30)
+	addtimer(CALLBACK(src, .proc/pulse, 0), 20)
```
Comment (Rohesie) [inline code/modules/assembly/voice.dm:85]:
```suggestion
	addtimer(CALLBACK(src, .proc/pulse), 20)
```
`FALSE` is the default value

**Comment by Winterous** [comment]
> 
> 
> Change all signalers to have a 3 second delay. Even mousetraps and proxy signalers work fine after this change, since once the process starts it runs its course and the bomb will go off.
> 
> Optional have a custom tune play once a signaler is activated such as that one popular nasheed people were raving about a few years ago.
> 
> Directly pinging the wire with a multitool would make a bomb go off instantly.
> 
> Introduce syndicate signalers that can be adjusted between instant and 3 second delay.

I think you're misunderstanding something here.
Grenades are no longer DETONATED when the wire is pulsed, they are PRIMED. Which means their usual fuse of 3-5 second in the case of chemical grenades will apply.

**Comment by Jared-Fogle** [comment]
> As fun as mouse traps are, as they are right now they might be a tiny bit too strong. Like, put nuke disk in a box, put a mousetrap bomb and several more bombs for good measure inside... Op reaches in, and becomes gibs. 

This is so common that I regularly see ops shoot open boxes or dump it.

File: code/modules/assembly/voice.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/assembly/voice.dm:25]: Ahhhh I'm a moron. scuse me....

File: code/modules/assembly/voice.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/assembly/voice.dm:85]: :)

File: code/modules/assembly/voice.dm [hunk 3]
```diff
@@ -76,6 +78,11 @@
 			if(length(raw_message))
 				. = TRUE
 
+/obj/item/assembly/voice/proc/send_pulse()
+	visible_message("clicks", visible_message_flags = EMOTE_MESSAGE)
+	playsound(src, 'licensed_sound/Sampling-Plus-1.0/whirthunk.ogg', 30)
+	addtimer(CALLBACK(src, .proc/pulse), 20)
```
Comment (Rohesie) [inline code/modules/assembly/voice.dm:84]:
```suggestion
	addtimer(CALLBACK(src, .proc/pulse), 2 SECONDS)
```
time macro

**Comment by Rohesie** [comment]
This needs someone to approve or deny on balance

**Comment by optimumtact** [comment]
I approve it

File: code/modules/assembly/voice.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/modules/assembly/voice.dm:84]: o7

**Comment by Rohesie** [comment]
Then I'll merge it once the change request is done.

**Comment by Jared-Fogle** [comment]
Just to be certain I've got everything understood, the concern here is being able to selectively choose when the grenade goes off in crit, yes? I get that. If I made a PR to bring back pulsing setting off grenades, but kept the part of voice analyzers not working with whispers (the only way to talk in crit), would that be acceptable?

Basically, I want mouse trap grenades back but without having to have a bunch of checks specifically for them.

**Comment by LemonInTheDark** [comment]
Instant anything bothers me really, but I can't really speak to that

**Comment by Winterous** [comment]
> Basically, I want mouse trap grenades back but without having to have a bunch of checks specifically for them.

Making them work like proxy grenades do would be my go-to, but I dunno if that's like, inefficient or anything. Honestly I think ANY wire attachment should override regular grenade behaviour in some way, like timers and proxies. Make them only function when primed.

**Pull request merged**